### DISCUSSION TITLE
Add support for TTL

### DISCUSSION THREAD

**Discussion by skrtdev** [spatie/once#68] [Ideas]
It would be nice to have caches that expire after a set time

**Comment by jeffreyvanhees** [comment]
The function is only cached within the current request, so a TTL seems unnecessary.

**Reply by skrtdev** [reply]
What about long-running processes?

**Reply by freekmurze** [reply]
People who need that can flush the cache manually: `Spatie\Once\Cache::getInstance()->flush();`

**Reply by skrtdev** [reply]
If i could use it it wouldn’t have asked for ttl
